### ISSUE TITLE
Parameters: { delta } might not be used, but delta is a parameter for a custom plugin

### ISSUE THREAD

**Issue by craigching** [dmlc/xgboost#6954]
We've written our own Huber loss function as an xgboost plugin. Our Huber loss uses a parameter called `delta` which combines L1 and L2 regularization terms on weights for our loss function (as described in wikipedia for the pseudo huber loss function). The problem is that xgboost complains about this parameter, saying that it may not be used. I would like it if xgboost could recognize plugin parameters and not issue this warning. I like the warning in general, but just not for our plugin parameters.

**Comment by trivialfis** [comment]
Hi, there's an example in the plugin in directory in xgboost source tree, inside that example you can see these 2 functions:

``` c++
  void SaveConfig(Json* p_out) const override {
    auto& out = *p_out;
    out["name"] = String("my_logistic");
    out["my_logistic_param"] = ToJson(param_);
  }

  void LoadConfig(Json const& in) override {
    FromJson(in["my_logistic_param"], &param_);
  }
```

It's for both serialization and parameter validation.  You need to implement them with the parameter named to something like:

```
  out["huber_param"] = ToJson(param_);  // notice the "_param" postfix.
```

**Comment by craigching** [comment]
Just want to report back that this worked. Thanks for the help!
